test/sharness/t0100-name.sh

```diff
@@ -225,6 +225,15 @@ test_name_with_self() {
         grep "argument \"ipfs-path\" is required" curl_out
         '
 
+        # test publishing with TTL
+        test_expect_success "'ipfs name publish --ttl=30m' succeeds" '
+        ipfs name publish --ttl=30m --allow-offline "/ipfs/$HASH_WELCOME_DOCS" && 
+        ipfs routing get "/ipns/$PEERID" > ipns_record &&
+        ipfs name inspect --verify $PEERID < ipns_record > verify_output &&
```

Comment:
Need additional check that reads `value` from DAG-JSON in `data` field, and confirms it matches the `value` from the top level

Reply:
Not exactly sure how to do that, as DAG-JSON has the field encoded in bytes so what you get is something like this:

```json
{
  "Entry": {
    "Value": "/ipfs/QmYuSqUfe8WdBKSCfCrbdqjiZtJgFSK75Q9MqBBabAdwcq",
    "ValidityType": 0,
    "Validity": "2023-01-26T07:45:20.787364Z",
    "Sequence": 0,
    "TTL": 1800000000000,
    "PublicKey": "m",
    "SignatureV1": "mqaPGy+Be71NDYIvcHg2rRL9Bqgvxb2dzIAoRm7YbJdNggYlGukMLLfF00FYAf0Fvg6UM2W4XfM0tleTIHHH5BA",
    "SignatureV2": "mOTUgjmKyBBKC75x4+S151XeL/QVTyurxplH9pGSkXPJ/gyWSu2ygcIhEWgmZWS9f5Wp/PeK2o/5XWK7YMStqCg",
    "Data": {
      "Sequence": 0,
      "TTL": 1800000000000,
      "Validity": {
        "/": {
          "bytes": "MjAyMy0wMS0yNlQwNzo0NToyMC43ODczNjRa"
        }
      },
      "ValidityType": 0,
      "Value": {
        "/": {
          "bytes": "L2lwZnMvUW1ZdVNxVWZlOFdkQktTQ2ZDcmJkcWppWnRKZ0ZTSzc1UTlNcUJCYWJBZHdjcQ"
        }
      }
    }
  },
  "Validation": null
}
```

Also, since the data field can be customised by the entry creator, I'm not sure if it's worth checking the value it can contain anything else in the real world.